
\chapter{Introduction}
\label{chapter: /Literature review on pharmacogenomics approaches, exome sequencing technolgy and variamt calling, Rare and common variant association tests, Quantitative and Qualitative association of the variants, Pathway based studies}

\section{Pharmacogenomics and Lung Cancer}
Lung cancer is the most lethal of all the cancer types. According to World Heath Organization (WHO) \cite{WHO} fact sheet of 2015, lung cancer caused 1.56 million death worldwide in 2012. With the overall survival rate of $18\% $, it was estimated $26\%$ of the all cancer deaths in 2014 and thus the leading cause of cancer death in the USA.\cite{siegel2014cancer}.

Thus chemotherapy with the standard platinum agents are the frequently administered to the patients with the advanced lung cancer.\cite{shiraishi2010association} Platinum based drugs such as  cisplatin, carboplatin and oxaliplatin are widely used. Platimium based agents forms the DNA adducts that thwarts cellular process and lead to apotosis. \cite {chen2013platinum}.The standard chemotherapeutic treatment for lung cancer are based on using platinum based agents with other another agents.\cite{goffin2010first} Microtubule - targeting agents such as paclitaxel, docetaxel,or vinorelbine and DNA-damaging agents such as gemcitabine or irinotecan are paired with the platinium-based agents in chemotherapeutic treatment.\cite{shiraishi2010association} 

Chemotherapy are provided to the patients in various regimen and doses. The chemotherapy regimen are administered based on the somatic  mutation cancer profiles and are aimed thwarting the cell growth and genomic integrity in the cancer cells. However, based on the genetic make of the individuals these chemotherapeutic drugs have reported to induce various adverse reaction mechanism , sometimes causing the death of the individuals.  

Pharmacogenomics conceptualizes the interaction between the human genetic components and the drug that are affected in various drug uptake mechanisms such as pharmacokinetics and pharmodymanics. Pharmacogenomics research aims at identifying genes and the gene variants involved in the interaction between the drugs and body. Genetic variants can greatly influence the nature of the effects a drug will  have on an individual as well as the amount of the drug required to produce the desired effect. In regards with the pharmacogenomics the genome wide association studies considers the traits as the drug dose dependent responses or the adverse event profiles.Association methods are used to discover novel associations between the drugs and genes on the cases and control samples. /paper1/ Pharmacogenomics study goal to identify the genetic variants and the underlying biology of the associated genes, /cite2 /

The main aim of the pharmacogenomics is to faciliatate the physicians witht the optimal drug selection, dose and treatment duration. /paper 5/

Pharmacogenomics have the potential to elucidate the adverse as well as positive effect of a drug have, based on the genetic make-up of an individuals. This understanding will help in the correct dosing, effective treatment strategies for various human diseases. Specifically these pharmacogenomics approaches are taken towards cancer and neurological disorders. In cancer, chemotherapeutic drugs targets the cellular mechanisms that halts the cancer growth, these could also have adverse reaction towards the other normal cells as well. With the advanced of the modern genotyping technologies from microarray based genotyping to massively parallel DNA sequencing provides the unprecedented potential to interrogate the nucleotide to single base-pair level.

\begin{itemize}
\item Lung cancer and pharmacogenomics
\item Genome-wide association studies using exome-sequencing approach
\item Mapping and variant calling of the sequenced data $-$GATK pipeline
\item Candidate gene based association studies
\item Single variant bases association studies
\item Gene$-$based association studies
\item Network based functional assignment of the associated genes
\end{itemize}


\section{GATK-pipeine Methods}
The phenotype information of the patients
GATK pipeline 
Assocaition methods used

Results

Discussion

Future works 

Reference


You should use transition in your text, meaning that you should help
the reader follow the thesis outline. Here, you tell what will be in
each chapter of your thesis. 

Mutations associated with the pharmacokinetics and pharmcodynamics affects the drug metabolisma and causes the adverse reaction in the patients. Germline pharmacogenomics can identify the patients at the highest risk of developing drug adverse response mechanisms. example TCL1A gene mutations are associated with the musculoskeletal adverse reaction with the patients receving aromatase inhibiotor /paper 3/.
 Another example is the thiopurine-S- methytransferase (TPMT ) and mercaptopurine (6-mercaptopurine).Variant in mercapthopurine reduces the enzyme function that are associated with the risk of severe mylosupressin / cite paper 4/
 Interested in the discovery of the genetic markers that are responsible for the drug response or toxicity as the chemotherapeutic drugs are have high toxicity indeces .  

Cancer drugs have many toxic effects and in many cases of advanced cancers, physicians guess and test medications by prescribing and monitoring them.

Pharmacogenomics provides the opportunity to predict the outcomes of the drugs usage in the treatment regimen of the cancer patients which helps in the individualized tailored oncology treatment of the patients.

The germline variation within the patient influences the pharmacokinetics and pharmacodynamics properties of the drugs. Pharmacokinetics - efffect if the body on the drug: process by which the drug is absorbed, distributed, metabolized and eliminated in the body. Pharmacodynamics- the effect of the drugs on the body that is the drug targets and mechanism of action.

New genomic technology and statistical methods are influencing the pharmacogenomic marker discovery

 
\section{Statistical Association studies of genotype and phenotype}

Single Site Case/Control tests
Linear and Logistic regression model for single variant
Gene-based tests 


Statistical Analysis strategies for association studies involving rare variants



Burden- based statistical tests : the test for the association between the variants and the continuous phenotype  consider a liner model given as:

Gene-based tests 


Statistical Analysis strategies for association studies involving rare variants



Burden- based statistical tests : the test for the association between the variants and the continuous phenotype  consider a liner model given as:

 \begin{equation*}
  Y_i = \alpha_0+ {a}^{'} X_i + {\beta}^{'} G_i+ {\epsilon}_{i}
  \end{equation*}
 
 where  
${\epsilon}_{i} \sim  N(O, {\sigma} ^2)$  and $\alpha_0$ is an intercept term , $\alpha = {[  {\alpha}_{1}, {\alpha}_{2},\dotsc,{\alpha}_{N} ]} ^ {'}$ is the vector of regression coefficient of the $m$ covariates and $\beta = [{\beta}_{1}, {\beta}_{2},\dotsc,{\beta}_{N}] ^{'}$

An adaptation of CMC burden test that collapses genotype information by counting the number of the variants in the region before applying logistic regression to the collapsed statistic.

 
 
 
 
 
 
 
 
 
 
 \chapter{Result}
\label{chapter: result}

\section{Summary Statistics of the study cohort clinical data}

In the current study, we sequenced the exomes of 216 samples. The study cohort consists of the Non small cell lung carcinoma (NSCLC) patients treated with theCarboplatin and Gemcitabine  chemothereupaetic drugs. The phenotypic characters of the lung cancer patients are shown in the table below. 

\begin{table}[thb]
	\centering
	\begin{tabular}{lll}
		\hline
		\textbf{Clinical Features} &  & \textbf{Patients} \\ \hline
		\multicolumn{3}{|l|}{\textbf{Gender of the Samples}} \\ \hline
		\multicolumn{1}{|l|}{} & \multicolumn{1}{l|}{Female} & \multicolumn{1}{l|}{115} \\ \hline
		\multicolumn{1}{|l|}{} & \multicolumn{1}{l|}{Male} & \multicolumn{1}{l|}{101} \\ \hline
		\multicolumn{3}{|l|}{\textbf{Age of treatment}} \\ \hline
		\multicolumn{1}{|l|}{} & \multicolumn{1}{l|}{Overall} & \multicolumn{1}{l|}{64.5 (60-71)*} \\ \hline
		\multicolumn{1}{|l|}{} & \multicolumn{1}{l|}{Female} & \multicolumn{1}{l|}{64 (59.50-70.00)*} \\ \hline
		\multicolumn{1}{|l|}{} & \multicolumn{1}{l|}{Male} & \multicolumn{1}{l|}{67 (61-72)*} \\ \hline
		\multicolumn{3}{|l|}{\textbf{Histological Subtype}} \\ \hline
		\multicolumn{1}{|l|}{} & \multicolumn{1}{l|}{Adenocarcinoma (AC)} & \multicolumn{1}{l|}{133} \\ \hline
		\multicolumn{1}{|l|}{} & \multicolumn{1}{l|}{Squamous Cell Carcinoma (SCC)} & \multicolumn{1}{l|}{41} \\ \hline
		\multicolumn{1}{|l|}{} & \multicolumn{1}{l|}{Large Cell Carcinoma (LCC)} & \multicolumn{1}{l|}{10} \\ \hline
		\multicolumn{1}{|l|}{} & \multicolumn{1}{l|}{Non-Small Cell Lung Cancer} & \multicolumn{1}{l|}{31} \\ \hline
		\multicolumn{3}{|l|}{\textbf{Smoking History}} \\ \hline
		\multicolumn{1}{|l|}{} & \multicolumn{1}{l|}{Current} & \multicolumn{1}{l|}{95} \\ \hline
		\multicolumn{1}{|l|}{} & \multicolumn{1}{l|}{Former} & \multicolumn{1}{l|}{100} \\ \hline
		\multicolumn{1}{|l|}{} & \multicolumn{1}{l|}{Never} & \multicolumn{1}{l|}{21} \\ \hline
		\multicolumn{3}{|l|}{\textbf{Pathological Stages}} \\ \hline
		\multicolumn{1}{|l|}{} & \multicolumn{1}{l|}{Stage Ia/Ib} & \multicolumn{1}{l|}{40} \\ \hline
		\multicolumn{1}{|l|}{} & \multicolumn{1}{l|}{Stage IIa/IIb} & \multicolumn{1}{l|}{29} \\ \hline
		\multicolumn{1}{|l|}{} & \multicolumn{1}{l|}{Stage IIIa/IIIb} & \multicolumn{1}{l|}{64} \\ \hline
		\multicolumn{1}{|l|}{} & \multicolumn{1}{l|}{Stage IV} & \multicolumn{1}{l|}{81} \\ \hline
		\multicolumn{3}{|l|}{\textbf{Treatment Type}} \\ \hline
		\multicolumn{1}{|l|}{} & \multicolumn{1}{l|}{Advanced disease} & \multicolumn{1}{l|}{142} \\ \hline
		\multicolumn{1}{|l|}{} & \multicolumn{1}{l|}{Adjuvant treatment} & \multicolumn{1}{l|}{74} \\ \hline
	\end{tabular}
	\caption{Clinical Features of Study Cohort. The figures in the right indicate the number of patients with the clinical features. The patients with $^*$ indicate the median age of the patients with the inter-qualtile range in the brackets}
	\label{Clinical-features-tables}
\end{table}
\newpage
\clearpage

The study cohort consists of equal proportion of male and female patients. The $61\%$ of the patients in the cohort has the adenocarcinoma histological subtype. And $85\%$ of patients in the study cohort have the smoking history of either being a current smoking status($43\%$) or former smoking history(($46\%$)). Similarly, most of the patients ($70\%$) are in advanced stages -  IIIa/IIb/IV of the lung cancer prognosis.   

%%%%%%%%%%%%%%%%%%%
\begin{figure}[h]
	\begin{center}
		\includegraphics[width=\textwidth]{Genderandpathology.pdf}
		\caption{The Gender and the lung cancer phenotype of the study cohorts. The barplot depicts the number of the female patients are higher than the male patients with the most of the patients are treated for advanced treatment.However the proportion of the patents with the different stages of the lung cancer are similar in both the sexes in the study cohort.}
		\label{Genderpathology}
	\end{center}
\end{figure}
%%%%%%%%%%%%%%%%%%

\section{Transformation of the Nadir TPK, LPK and NPK }

This is a test files.This is a test files.
This is a test files.
This is a test files.
This is a test files.
This is a test files.

\begin{figure}[h]
	\begin{center}
		\includegraphics[width=\textwidth]{NadirvsBaseline.pdf}
		\caption{The figure illustrates the decrease in the count of the thrombocytes, leucocytes and neutrophil after the patients have been treated with the chemotherapeutic drugs}
		\label{NadirVSBaseline}
	\end{center}
\end{figure}

\section{Results from the exome sequencing}
\section{Quality Control of the VCF files}
\section{Post Processing VCF file}
\section {Statistical Association of single variants from Quantitative and Qualitative analysis}
\section{Statistical Association from Gene based tests}




\chapter{Methods}

This chapter covers the overall piple-line  and the methods used in the thesis project. The work-flow of the overall pipeline is described below in the Figure~\ref{Chart of project} . Each step discusses the pipeline followed in the project.  
  
  %% Drawing flow chart of the process
  %% Define block styles
  
\tikzstyle{block} = [rectangle, draw, fill=blue!10, text width=4.5em, text centered, rounded corners, text width=20em, minimum height=3em]
\tikzstyle{block2} = [draw, fill=green!10, text width=8.5em, text centered, rectangle, node distance=2cm,minimum height=3em]
\tikzstyle{block3} = [draw, fill=green!10, text width=8em, text centered, rectangle, node distance=6.5cm,minimum height=3em]
\tikzstyle{line} = [draw, -latex']
  
\begin{figure}[ht]
  \centering
\begin{tikzpicture}[node distance = 2cm, auto]
  % Place nodes
\node [block] (First) {Study Cohort};
\node [block, below of=First] (Second) {Whole Exome-Sequencing Of Study Cohort};
\node [block, below of=Second] (Third) {Preprocessing/quality control of Sequencing Reads};
\node [block, below of=Third] (Fourth) {Mapping, Alignment and Variant Calling of Sequencing Reads};
\node [block, below of=Fourth] (Fifth) {Post processing/Quality control of VCF files};
\node [block2, below of=Fifth] (Sixth) {Case/Control Based Association Studies in Extreme Phenotypes};
\node [block3, left of=Fifth] (Seventh) {Candidate Gene based Studies in Extreme Phenotypes};
\node [block3, right of=Fifth] (Eight) {Quantitative Trait Association Studies};  
  
  % Draw edges
\path [line] (First) -- (Second);
\path [line] (Second) -- (Third);  
\path [line] (Third) -- (Fourth);
\path [line] (Fourth) -- (Fifth);
\path [line] (Fifth) -- (Sixth);
\path [line] (Fifth) -- (Seventh);
\path [line] (Fifth) -- (Eight);
  
\end{tikzpicture}
	 \caption{Flow-chart of project}
	\label{Chart of project}
\end{figure}  
  
 \section{Study Cohort}

A total of 217 patient diagnosed with Non-small cell lung cancer were included in the study. All patients were scheduled to be treated with carboplatin and gemcitabine for four cycles in and received at least one cycle of carboplatin and/or gemcitabine chemotherapy. After the chemotherapy cycle, the nadir values of the myelosupression are monitored and blood count of leucocytes, neutrophils, platelets and haemoglobin are observed. Based on the observed nadir values of the individual myelosupression, the patient cohort were graded as 0,1,2,3 or 4 based on the Common Toxicity Criteria (CTC) \cite{trotti2000common} grade set up by National Cancer Institute (NCI). 

\textbf{SELF NOTE Can dosing measurement of the patients could be added to the data?}


\section{Whole-exome Sequencing of the patient cohort}

Whole exome sequencing involves the sequencing of the protein coding region of the genome and involves the total of $2\%$ of whole genome.In the current project, DNA from the patient sample was extracted from the whole blood and sequenced in Nextera\textsuperscript{\textregistered} Rapid Capture Exomes kit. The sequencing of the individual samples  were performed in Illumina\textsuperscript{\textregistered}Hiseq2000 platform to generate read lengths of $2 \times 150$ base pairs. The exome sequencing was done in the National Genomics platform at Science for Life Lab, Solna.

In the current project I worked with the GATK pipelines for the three samples. However, following the final VCF files the quality control and subsequent statistical analysis was performed in all the study cohort which was obtained with the identical GATK pipelines.

\section{Preprocessing/quality control of Sequencing Reads}

From the fastq files generated from the Illumina pipelines, the quality control pipeline was applied before mapping and aligning to the reference genome. The quality control and adapter removal was done with the utility program Trim Galore\cite{TrimGalore}. TrimGalore is programming script, that makes use of Cutadapt \cite{Martin2011Cutadapt} to trim Illuminia adapter sequences and low quality ends with quality threshold of 25 on Phred scale and only trim reads greater than 25 for each read. A FastQC report of each samples is generated on the trimmed sequences.

\section{Mapping, Alignment and Variant Calling of Sequencing Reads}

The post trimmed sequence reads are then processed to further mapping and alignment quality control measures. The best practise GATK-pileline \cite{depristo2011framework} was used for aligning and mapping of sequence reads. The basic work flow of the GATK pipeline is shown in the figure below. 

$  Add figur please $

First of all before processing into the GATK pipeline, the fastq files trimmed from  the Illuminia pipelines are aligned to GRCh37/hg18 human reference genome (UCSC Genome Browser) with Burrows - Wheeler Aligner (bwa/0.5.9) \cite{li2010fast}. All reads were mapped to the positive strand of the hg19. The command BWA MEM was used to align the reads as the read length were greater than 100 base pair.The resulting sam files were converted bam files using samtools \cite{li2010fast}. Using the samtools, we also filtered the reads that are mate reverse strand and optical duplicates using suppling -F 1056 to the samview commands. Thus obtained bam files were processed with MarkDuplicates, Picard (cite picard website) tools to mark the duplicate reads from the mapped reads. 

Post processing of these aligned reads was done using GATK (v3.3-0) where the reads were subjected to indel realignment and base-quality recalibration. Variant detection and genotyping was done on HaplotypeCaller from GATK and performed on the targeted regions in an individual sample. Variant calls from the individual samples was stored as raw gvcf file. HaplotypeCaller estimate the probability that a given site is non-variant.The individual gvcfs are collectively collected and formatted to generate the multisample Variant call Format (VCF) file. The raw variants in the VCF files are flagged if thy had the quality scores $<$ 50,FisherSB filter $>$ 60, quality by depth $<$ 5. Thus, obtained VCF file is termed as master raw vcf file which contain information of all the samples in the study. Finally, all the variants in VCF file was annotated using SNPEff \cite {cingolani2012program} which annotates and predicts the effects of variants on genes (such as amino acid changes).

\section{Post processing/Quality control of VCF files}

Individual exome sequencing data and variants were evaluated against the various quality control metrics such as reads counts mapped to hg19, reads mapped to the target, transition/transversion (Ti/Tv) ratio, percentage of sites covered at 1X, 5X, 10X, 20X, 30X coverage. Picard command $CalculateHsMetrics$ was used to calculate the quality metrices.  

The raw VCF file was filtered using the vcftools\cite{danecek2011variant} based on the individual quailty of the variants output by GATK pipeline. We only kept the variants which have passed the GATK filter and have the mean sequencing depth $\geq$ 10.  

In order to confirm the ancestry of study group, we carried out in plink identity based on descent clustering  of the samples using Idenity by descent tool (IBD) tool in plink \cite{purcell2007plink}. We found similarity between the S0922 and S1056 and with further confimation from the clinical source these samples were identical. Thus sample S0922 was removed from further analysis. On the resulting patient cohorts we further filtered the samples with the low genotyping rate and kept the variants with genotyping rate of at least 95\% for the samples. Samples S0328, S0664 were removed as they had genotyping rate lower than 95\% as shown in the figure.... Similarly, sample S0580 was removed there was number of variants were subtantial higher than other samples which suspected of contamination of sample. The resulting $212$ samples's VCF were termed as master VCF file. From the master VCF file, we classified the variants as Common and Rare based on the threshold of $0.01$ MAF in the study cohort.  

subsection{Filtration of the VCF files}

(Add the picture.. )
 
\section{Quantitative Trait Association Studies}
\label{Quantitative Trait Association studies}

% This method is among the different methods I have used. But to start off the writing I have startes with this thing and lets hope I could write something today so that I could write and edit all the things 

%Flow chart of the overall process ... Needs to be done

In this section, I discuss the approaches that were taken in association the genetic variations with the quantitative phenotypes that is nadir values of the leucopenia, neutropenia, and thrombopenia. The intial measurement of the patients Leucocytes, Thrombocytes and Neutrophils count were measured before and after the chenmotherapy treatment. The quantiative trait analysis assumes the phenotype to be normally distributed, however from the intial histogram plot as shown in the figure below showed the count data were skewed for each phnotypes. Hence, we applied two transformation method to normalize the individual nadir values for the each phenitypes. For each toxicity phenotypes for the cancer patients we first did the $\log$ conversion of the individual values. Next, we applied the Empricial Normal Quantile Transformation(ENQT) method to normalized the data. We used the R-package multic \cite{multic} and command $t.rank$ to normalized the data. The normality of the phenotype data was tested using the $shapiro.test$ in the stats package in R \cite{Rprogram} for each transformed and untransformed data. The missing values were in the phenotypes were labelled as $-9$. These $\log$ and ENQT transformed data were then used as quantitative phenotype values in the association analysis in plinkseq\cite{Doe:2009:Online}. 


\subsection{Single Variant Association Tests}

 The Single Variant Association study was performed on bi allelic variants with the MAF $> 0.01$ in the study cohort. We performed the unadjusted linear association for the all common variants in the study cohorts. The linear regression was performed using $-assoc$ command in plink \cite{purcell2007plink}. The linear regression association was done for the both log-transformed and ENQT transformed data for all the filtered samples. he Single nucleotide polymoprhisms(SNP) with p-value $<$ 0.01 after multiple correction was considered as significant. 

\subsection{Gene/Region Based Association Tests}

For the gene/region based association, we investigated the effect of the both the rare and common variants of the genes to the toxicity phenotype. Thus we used the master VCF file with all the variants for association tests.

\subsubsection{Gene/Region definition for association studies}

Initially, we defined the regions of genes considered for the association studies.The genes in the VCF files were annotated as per Refseq \cite{pruitt2014refseq} and USSC Genome Browser \cite{karolchik2003ucsc} databases. The longest transcript annotation was considered as the standard genes. Further, we were specifically segmented the genes into smaller exons $\pm$ six base-pairs regions for the association studies. This approach decreased the number the variants in more than one genes, and included the splice variants within the genes.

\subsubsection{Gene/Region based Association test}

The bi-allelic variants were mapped to the associated exons using $gene-report$ command in plink. The mapped variants were then used for the association tests in SKAT \cite{SKAT} package. The log and ENQT transformed  quantitative data from the individual patients used for the association test. The implementation of gen based tests such as SKAT, Burden and SKATO are used for each of the phenotypes. Since the toxicity phenotype is complex traits we considered the equal contribution of common and rare variants to the phenotype. Thus we equally weighted both common and rare variants in association tests. The genes with p-value $<$ 0.01 after multiple correction was considered as significant. 


\section{Case/Control Based Association Studies in Extreme Phenotypes}
\label{Qualitative Trait Association studies}

We also looked in Qualitative study design for the association studies. The Common Toxicity Criteria (CTC) score of the individual phenotype was considered to classify the patients into high or low toxicity group.

\subsection{Definition of extreme cases and control from the study Cohort}

For each of the inividual myelosupression toxicity phenotypes we classified patients  as high toxicity (Cases)with the CTC score of either 3 or 4 and as low toxicity Control group of CTC score 3 or 4.The study cohort samples of patients in each group for each phenotype is shown in the Table below.

\begin{table}[h]
	\begin{tabular}{lll}
		\hline
		\multicolumn{1}{|l|}{\textbf{Phenotype}} & \multicolumn{1}{l|}{\textbf{High Toxicity(Cases)}} & \multicolumn{1}{l|}{\textbf{Low Toxicity (Control)}} \\ \hline
		\multicolumn{1}{|l|}{Thrombopenia (TPK)} & \multicolumn{1}{l|}{75} & \multicolumn{1}{l|}{92} \\ \hline
		\multicolumn{1}{|l|}{Leucopenia (LPK)} & \multicolumn{1}{l|}{49} & \multicolumn{1}{l|}{90} \\ \hline
		\multicolumn{1}{|l|}{Neutropenia (NPK)} & \multicolumn{1}{l|}{97} & \multicolumn{1}{l|}{76} \\ \hline
\end{tabular}
\caption{Number of patients in each phenotype. The different toxicity phenotypes of the individual patients are given in the first column and the corresponding cases of High toxicity group with the CTC score of either $1$ or $0$ and control group of Low Toxicity with the CTC score of either $3$ or $4$ are tabulated in each successive columns.}
\label{phenotype classification}
\end{table}


\subsection{Single Variant Association Tests}

The Single Variant Association study was performed on bi-allelic variants with the MAF $> 0.01$ in the each phenotype case and control group. We performed the unadjusted logistic association for the all common variants. The logistic regression was performed using $-assoc$ command in plink \cite{purcell2007plink}. The Single nucleotide polymoprhisms(SNP) with p-value $<$ 0.01 after multiple correction was considered as significant. 

\subsection{Gene/Region Based Association Tests}

For the gene/region based association, we investigated the effect of the both the rare and common variants of the genes to the toxicity phenotype of both case and control group. Thus we used the master VCF file with all the variants for association tests.

\subsubsection{Gene/Region definition for association studies}

Initially, we defined the regions of genes considered for the association studies.The genes in the VCF files were annotated as per Refseq \cite{pruitt2014refseq} and USSC Genome Browser \cite{karolchik2003ucsc} databases. The longest transcript annotation was considered as the standard genes. Further, we were specifically segmented the genes into smaller exons $\pm$ six base-pairs regions for the association studies. This approach decreased the number the variants in more than one genes, and included the splice variants within the genes.

\subsubsection{Gene/Region based Association test}

The bi-allelic variants from the cases and control groups were mapped to the associated exons using $gene-report$ command in plink. The mapped variants were then used for the association tests in SKAT \cite{SKAT} package. We assigned the phenotype groups into binary labels such that High Toxicity groups was labelled as $1$ while the low toxicity control group was labelled as $0$. The implementation of binary gene based tests -SKAT, Burden and SKATO are used for each of the phenotypes. Since the toxicity phenotype is complex traits we considered the equal contribution of common and rare variants to the phenotype. Thus we equally weighted both common and rare variants in association tests. The genes with p-value $<$ 0.01 after multiple correction was considered as significant. Additionally, we used the default adjustment of the small- samples from the SKAT package in R.


  
